### ISSUE TITLE
CUDNN_STATUS_NOT_SUPPORTED for large matrix input

### ISSUE THREAD

**Issue by nikostr** [pytorch/pytorch#4107]
The following code (adapted from https://discuss.pytorch.org/t/cudnn-status-not-supported-error-occurs-when-apply-autograd-grad-to-compute-high-order-differentiation/8256) gives me a CUDNN_STATUS_NOT_SUPPORTED error. I'm running the pytorch 0.3.0 with cuDNN 7.0.4,  CUDA 9.0.176, and python 3.6. 

```import torch.nn as nn
from torch.autograd import Variable, grad
import torch.utils.data as Data

class TestDataset(Data.Dataset):

    def __init__(self):
        self.sequences = []
        PROBLEM_SIZE = 171 * 21
        data = torch.rand(1,PROBLEM_SIZE,PROBLEM_SIZE)
        label = torch.rand(PROBLEM_SIZE,PROBLEM_SIZE).round()
        self.sequences.append((data,label))

    def __len__(self):
        return len(self.sequences)

    def __getitem__(self,idx):
        return self.sequences[idx]

train_data = TestDataset()

train_loader = Data.DataLoader(
    dataset=train_data, batch_size=1, shuffle=True, num_workers=1)


class CNN(nn.Module):
    def __init__(self):
        super(CNN, self).__init__()
        self.conv1 = nn.Conv2d(1, 1, kernel_size=3, stride=1, dilation=1, padding=1,bias=False)
        self.conv2 = nn.Conv2d(1, 1, kernel_size=3, stride=1, dilation=2, padding=2,bias=False)

    def forward(self, x):
        x = self.conv1(x)
        x = self.conv2(x)
        return x


cnn = CNN()
cnn.cuda()

loss_func = nn.BCEWithLogitsLoss()

for step, (data, label) in enumerate(train_loader):
    input = Variable(data).cuda()
    target = Variable(label).cuda()

    output = cnn(input)[0]
    loss = loss_func(output, target)

    params = cnn.parameters()
    g = grad(loss, params, create_graph=True)

    g_sum = 0
    for g_para in g:
        g_sum += g_para.sum()

    params = cnn.parameters()
    hv = grad(g_sum, params, create_graph=True)

    break
``` 

Running the convolutions individually works, as does running this code for smaller input matrices. I'm pretty sure the problem also occurs for sizes other than 3591 by 3591. The error I receive is the following:

```Traceback (most recent call last):
  File "/home/n/nikostr/pfs/CompBio-DD2402/experiments/20171209/mwe.py", line 54, in <module>
    g = grad(loss, params, create_graph=True)
  File "/home/n/nikostr/pfs/.conda/envs/myroot/lib/python3.6/site-packages/torch/autograd/__init__.py", line 158, in grad
    inputs, only_inputs, allow_unused)
RuntimeError: CUDNN_STATUS_NOT_SUPPORTED. This error may appear if you passed in a non-contiguous input.
```

I posted this as a question on the forums (https://discuss.pytorch.org/t/cudnn-status-not-supported-for-large-matrix-input/10986), and it has been reproduced.

cc @ezyang @gchanan @zou3519 @bdhirsh @heitorschueroff @csarofeen @ptrblck

**Comment by csarofeen** [comment]
I reproduced this with: 

```
import torch
import torch.nn as nn
from torch.autograd import Variable, grad

SIZE=[1, 1, 171*21, 171*21]
input = Variable(torch.cuda.FloatTensor(*SIZE).uniform_(), requires_grad=True)
conv1 = nn.Conv2d(1, 1, kernel_size=3, stride=1, dilation=1, padding=1,bias=False).cuda()
output = conv1(input)
loss = output.sum()
loss.backward()
```

I've confirmed it is a cuDNN bug. I've forwarded it to the cuDNN team and they will fix it as soon as possible given their development, testing and release cycle timelines.

**Comment by hkharryking** [comment]
I met the similar problem, is it solved?

**Comment by ngimel** [comment]
for now, you can try disabling cudnn
```torch.backends.cudnn.enabled = False```. Don't know what kind of performance you'll get, but better than erroring out.

**Comment by hkharryking** [comment]
Thanks, I tried that and my program has been runnable, but 10 times slower than the original.

**Comment by ngimel** [comment]
Can you please show minimum example that generates the error?

**Comment by nikostr** [comment]
@hkharryking as of cuDNN 7.0.5 there was no mention in the release notes of this being resolved. I had a look at this a couple of days ago, but I haven't actually attempted to upgrade and run with the latest version.

**Comment by ngimel** [comment]
@nikostr your case is not fixed.

**Comment by hkharryking** [comment]
Thank you all for the answers, I can reconstruct my problem by the following code.
my cudnn version is 7003

import torch
from torch.autograd import Variable
import torch.nn as nn

x = Variable(torch.rand(100000, 16,200).contiguous()).cuda()
bn = nn.BatchNorm1d(16)
bn.eval()      # add the eval() leads to the error, if it is commented, the programs will be ok.
bn.cuda()

x = bn(x)
print(x.size())

**Comment by nikostr** [comment]
@ngimel any idea when this might get resolved? :)

**Comment by fmassa** [comment]
To complement this thread, yet another case where CUDNN fails on backward with `CUDNN_STATUS_NOT_SUPPORTED.`:
```python
m = nn.Conv2d(1, 1, (3, 1), stride=(2, 1)).cuda()
i = torch.rand(131072, 1, 1537, 1).cuda()
o = m(i)
o.sum().backward()
```

I'm using CUDNN v7.0 with cuda 9.0.

**Comment by Jiaming-Liu** [comment]
any update?
my dirty solution is to add a dummy channel:
```
m = nn.Conv2d(2, 2, (3, 1), stride=(2, 1), group=2).cuda()
i = torch.rand(131072, 1, 1537, 1).cuda()
o = m(i.repeat(1,2,1,1))[:,:1,:,:]
```

**Comment by wasiahmad** [comment]
@Jiaming-Liu can you elaborate your dummy channel approach? How and why it works? I am also struggling with the error being discussed here.

**Comment by ngimel** [comment]
You can also try running with `torch.backends.cudnn.benchmark=True`

**Comment by wasiahmad** [comment]
@ngimel already tried but didn't solve the issue.

**Comment by Jiaming-Liu** [comment]
@wasiahmad I found this trick when using `Conv2dTranspose` with 1-channel in & 1-channel out.

CUDNN's Conv2dTranspose seems to hate 1-channel input/output. So dummy channel works.

**Comment by piercefreeman** [comment]
Do we know whether this is isolated to convolutional layers?  Or any other specifics on when we can expect to see this issue surface? @ngimel

**Comment by ezyang** [comment]
Yeah, if someone can work out exactly what situations this occurs, we can modify our kernel selection logic to not use cuDNN when parameters satisfy the problem.

**Comment by AlenUbuntu** [comment]
Still get the same error. When I use the model for testing, if I set batch size to be 1000, I get this error. However, if I set batch size to be 200, it works well.

**Comment by changdq12** [comment]
> I reproduced this with:
> 
> ```
> import torch
> import torch.nn as nn
> from torch.autograd import Variable, grad
> 
> SIZE=[1, 1, 171*21, 171*21]
> input = Variable(torch.cuda.FloatTensor(*SIZE).uniform_(), requires_grad=True)
> conv1 = nn.Conv2d(1, 1, kernel_size=3, stride=1, dilation=1, padding=1,bias=False).cuda()
> output = conv1(input)
> loss = output.sum()
> loss.backward()
> ```
> 
> I've confirmed it is a cuDNN bug. I've forwarded it to the cuDNN team and they will fix it as soon as possible given their development, testing and release cycle timelines.

@csarofeen Hi, I meet a similar problem if I set the batch_size to be bigger than 65535 in calling a batch normalization layer with mxnet. Have you fixed this problem? Will it help to update the version of cudnn?

**Comment by richard-vogl** [comment]
I am running into a similar problem. Does anyone know what causes this by now, or is there a workaround for this problem?

**Comment by Jiaming-Liu** [comment]
@richard-vogl try this:
> any update?
> my dirty solution is to add a dummy channel:
> 
> ```
> m = nn.Conv2d(2, 2, (3, 1), stride=(2, 1), groups=2).cuda()
> i = torch.rand(131072, 1, 1537, 1).cuda()
> o = m(i.repeat(1,2,1,1))[:,:1,:,:]
> ```
> 
> this run much faster than `torch.backends.cudnn.enabled = False`

**Comment by richard-vogl** [comment]
@Jiaming-Liu 

> @richard-vogl try this:
> 
> > any update?
> > my dirty solution is to add a dummy channel:
> > ```
> > m = nn.Conv2d(2, 2, (3, 1), stride=(2, 1), groups=2).cuda()
> > i = torch.rand(131072, 1, 1537, 1).cuda()
> > o = m(i.repeat(1,2,1,1))[:,:1,:,:]
> > ```
> > 
> > 
> > this run much faster than `torch.backends.cudnn.enabled = False`

Thanks for the response! In my case I already use several channels, I also tried to expand the singular batch dimension, but it doesn't really change anything.
Here my minimal code example to reproduce the problem:

```
import torch
x = torch.rand(1, 16, 70000, 100).to('cuda:2')
conv = torch.nn.Conv2d(in_channels=16, out_channels=32, kernel_size=(3,3), bias=False).to('cuda:2')
out = conv(x)
```
Since I have a stack of convolutions where i increase the channel size, and the problem occurs here, it fells like this is a problem linked to memory requirements - note that the total size of matrices exceeds 1GB with this convolution...

**Comment by ptrblck** [comment]
Hi @richard-vogl,

I couldn't reproduce the issue using your code snippet.
Could you post some information about your system (which GPU) and the PyTorch, CUDA, cudnn versions?

**Comment by richard-vogl** [comment]
> Hi @richard-vogl,
> 
> I couldn't reproduce the issue using your code snippet.
> Could you post some information about your system (which GPU) and the PyTorch, CUDA, cudnn versions?

Hi,

GPU: 
`GeForce GTX 1080 Ti (rev a1)`
pytorch:
```
> pip show torch
Name: torch
Version: 1.1.0
Summary: Tensors and Dynamic neural networks in Python with strong GPU acceleration
Home-page: UNKNOWN
Author: UNKNOWN
Author-email: UNKNOWN
License: UNKNOWN
Location: /home/rvogl/.local/lib/python3.6/site-packages
Requires: numpy
Required-by: torchvision
```
CUDA:
`NVIDIA-SMI 430.26       Driver Version: 430.26       CUDA Version: 10.2`

cudnn:
```
torch.backends.cudnn.version()
7501
```

**Comment by richard-vogl** [comment]
> Hi @richard-vogl,
> 
> I couldn't reproduce the issue using your code snippet.
> Could you post some information about your system (which GPU) and the PyTorch, CUDA, cudnn versions?

i can also reproduce it on those machines:

```
NVIDIA-SMI 418.67       Driver Version: 418.67       CUDA Version: 10.1   
GeForce GTX 1080

torch.backends.cudnn.version()
7402

Name: torch
Version: 1.0.1.post2
```

I **can not** reproduce it on this machine:

```
torch.backends.cudnn.version()
7402

NVIDIA-SMI 387.34                 Driver Version: 387.34
GeForce GTX TITAN Black
Tesla K40c

Cuda compilation tools, release 9.1, V9.1.85

pip3.6 show torch
Name: torch
Version: 1.0.1.post2
```

**Comment by blindpikachu** [comment]
Could it be related to lack of remaining memory on GPU device?

**Comment by richard-vogl** [comment]
> Could it be related to lack of remaining memory on GPU device?

definitely not lack of memory since there is more than enough memory available on the cards I used.
but I originally thought that it is somehow related to matrix size, see https://github.com/pytorch/pytorch/issues/4107#issuecomment-512336351

but as you can also see from the update/edit - it seems to be related more to the shape of the tensor than its size...
if someone wants to debug this and is unable to reproduce it: I might be able to setup an account on one of the machines where i can reproduce this - if it helps.

**Comment by felixlaumon** [comment]
@ptrblck I can reproduce this `CUDNN_STATUS_NOT_SUPPORTED` on `BatchNorm1d` on a Collab notebook (https://accounts.google.com/signin/v2/identifier?continue=https%3A%2F%2Fcolab.research.google.com%2Fdrive%2F1MMukaj-kfLAdvVWxI4ccsR07gaA90PFe&authuser=0&hl=en&flowName=GlifWebSignIn&flowEntry=ServiceLogin)

It seems `BatchNorm1d` / `cuDNN` has a upper limit batch size of `2 ** 16`.

```py
torch.__version__  # -> '1.3.0+cu100'
x = torch.rand((2 ** 16, 64, 60)).cuda().float()  # this will lead to an error in bn.forward
x = torch.rand((2 ** 16 - 1, 64, 60)).cuda().float()  # no errors for bn.forward
bn = nn.BatchNorm1d(64).eval().cuda()
x = bn.forward(x)
print(x.shape)
```

```
> nvcc --version
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2018 NVIDIA Corporation
Built on Sat_Aug_25_21:08:01_CDT_2018
Cuda compilation tools, release 10.0, V10.0.130
```

```
> nvidia-smi
NVIDIA-SMI 430.40       Driver Version: 418.67       CUDA Version: 10.1 
```

This large amount of `batch_size` is required in my use case due to batched sparse pointcloud input.

Hope this information helps! Are the workarounds you can recommend for now? (except disabling CuDNN)

**Comment by wbhu** [comment]
Hi @felixlaumon, 

I am also play with sparse point cloud and meet the same problem. When I disable torch.backends.cudnn during eval phase, the code is runable, but the BN in eval() model acts really strange, especially when worked with syncBN. Have you meet something like this?

Thanks very much

**Comment by felixlaumon** [comment]
@wbhu as a workaround you might want to try the non-CuDNN version of batch norm and wraps it as a layer. Below is an example for non-CuDNN BN in 1D

```py
def no_cudnn_batch_norm(
    input,
    running_mean,
    running_var,
    weight=None,
    bias=None,
    training=False,
    momentum=0.1,
    eps=1e-5,
):
    """F.batch_norm with cuDNN disable"""
    if training:
        size = input.size()
        size_prods = size[0]
        for i in range(len(size) - 2):
            size_prods *= size[i + 2]
        if size_prods == 1:
            raise ValueError(
                "Expected more than 1 value per channel when training, got input size {}".format(
                    size
                )
            )

    return torch.batch_norm(
        input, weight, bias, running_mean, running_var, training, momentum, eps, False
    )


class BatchNorm1dNoCuDNN(nn.modules.batchnorm.BatchNorm1d):
    """BatchNorm1d with CuDNN disabled"""

    def forward(self, input):
        # All of below is copy and pasted. Just replaced F.batch_norm with
        # no_cudnn_batch_norm
        self._check_input_dim(input)
        if self.momentum is None:
            exponential_average_factor = 0.0
        else:
            exponential_average_factor = self.momentum

        if self.training and self.track_running_stats:
            # TODO: if statement only here to tell the jit to skip emitting this when it is None
            if self.num_batches_tracked is not None:
                self.num_batches_tracked += 1
                if self.momentum is None:  # use cumulative moving average
                    exponential_average_factor = 1.0 / float(self.num_batches_tracked)
                else:  # use exponential moving average
                    exponential_average_factor = self.momentum

        return no_cudnn_batch_norm(
            input,
            self.running_mean,
            self.running_var,
            self.weight,
            self.bias,
            self.training or not self.track_running_stats,
            exponential_average_factor,
            self.eps,
        )
```

**Comment by divyanshj16** [comment]
This is modified code from @hkharryking .
```python
import torch
from torch.autograd import Variable
import torch.nn as nn

x = torch.rand(65535, 16,200).contiguous().cuda()
bn = nn.BatchNorm1d(16)
bn.eval() # add the eval() leads to the error, if it is commented, the programs will be ok.
bn.cuda()

x = bn(x)
print(x.size())
``` 
Output: torch.Size([65535, 16, 200])

If I increase batch size by 1.

```python
import torch
from torch.autograd import Variable
import torch.nn as nn

x = torch.rand(65536, 16,200).contiguous().cuda()
bn = nn.BatchNorm1d(16)
bn.eval() # add the eval() leads to the error, if it is commented, the programs will be ok.
bn.cuda()

x = bn(x)
print(x.size())
```
Output:
```
~\.conda\envs\pytorcharcgis\lib\site-packages\torch\nn\modules\module.py in __call__(self, *input, **kwargs)
    491             result = self._slow_forward(*input, **kwargs)
    492         else:
--> 493             result = self.forward(*input, **kwargs)
    494         for hook in self._forward_hooks.values():
    495             hook_result = hook(self, input, result)

~\.conda\envs\pytorcharcgis\lib\site-packages\torch\nn\modules\batchnorm.py in forward(self, input)
     81             input, self.running_mean, self.running_var, self.weight, self.bias,
     82             self.training or not self.track_running_stats,
---> 83             exponential_average_factor, self.eps)
     84 
     85     def extra_repr(self):

~\.conda\envs\pytorcharcgis\lib\site-packages\torch\nn\functional.py in batch_norm(input, running_mean, running_var, weight, bias, training, momentum, eps)
   1695     return torch.batch_norm(
   1696         input, weight, bias, running_mean, running_var,
-> 1697         training, momentum, eps, torch.backends.cudnn.enabled
   1698     )
   1699 

RuntimeError: cuDNN error: CUDNN_STATUS_NOT_SUPPORTED. This error may appear if you passed in a non-contiguous input.
```

**Comment by ngimel** [comment]
Please let's track batchnorm related issues in #29744, original issue here was for convolution.

**Comment by michaelklachko** [comment]
Just ran into this error while trying to train EfficientNet with batch size 512 on a single GPU. 

To reproduce, clone https://github.com/rwightman/pytorch-image-models repo and run this command: 
`./distributed_train.sh 1 /imagenet/ --model efficientnet_b0 -b 512 --sched step --epochs 450 --decay-epochs 2.4 --decay-rate .97 --opt rmsproptf --opt-eps .001 -j 8 --warmup-lr 1e-6 --weight-decay 1e-5 --drop 0.2 --drop-connect 0.2 --model-ema --model-ema-decay 0.9999 --aa rand-m9-mstd0.5 --remode pixel --reprob 0.2 --amp --lr .064`

On a single V100 it produces the following error right after it finishes the first epoch of training, either immediately before or during validation stage:

```
Training with a single process on 1 GPUs.
Model efficientnet_b0 created, param count: 5288548
Data processing configuration for current model + dataset:
        input_size: (3, 224, 224)
        interpolation: bicubic
        mean: (0.485, 0.456, 0.406)
        std: (0.229, 0.224, 0.225)
        crop_pct: 0.875
Selected optimization level O1:  Insert automatic casts around Pytorch functions and Tensor methods.

Defaults for this optimization level are:
enabled                : True
opt_level              : O1
cast_model_type        : None
patch_torch_functions  : True
keep_batchnorm_fp32    : None
master_weights         : None
loss_scale             : dynamic
Processing user overrides (additional kwargs that are not None)...
After processing overrides, optimization options are:
enabled                : True
opt_level              : O1
cast_model_type        : None
patch_torch_functions  : True
keep_batchnorm_fp32    : None
master_weights         : None
loss_scale             : dynamic
NVIDIA APEX installed. AMP on.
Scheduled epochs: 450
Train: 0 [   0/2502 (  0%)]  Loss:  6.952626 (6.9526)  Time: 21.453s,   23.87/s  (21.453s,   23.87/s)  LR: 1.000e-06  Data: 11.892 (11.892)
Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 32768.0
Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 16384.0
Train: 0 [1000/2502 ( 40%)]  Loss:  6.941184 (6.9479)  Time: 0.653s,  783.90/s  (0.870s,  588.41/s)  LR: 1.000e-06  Data: 0.074 (0.277)
Train: 0 [2000/2502 ( 80%)]  Loss:  6.913240 (6.9434)  Time: 0.647s,  791.28/s  (0.826s,  620.06/s)  LR: 1.000e-06  Data: 0.068 (0.239)
/home/mklachko/miniconda2/envs/pt/lib/python3.7/site-packages/PIL/TiffImagePlugin.py:804: UserWarning: Corrupt EXIF data.  Expecting to read 4 bytes but only got 0.
  warnings.warn(str(msg))
Train: 0 [2501/2502 (100%)]  Loss:  6.935285 (6.9416)  Time: 0.771s,  663.77/s  (0.828s,  618.38/s)  LR: 1.000e-06  Data: 0.193 (0.242)
Traceback (most recent call last):
  File "train.py", line 581, in <module>
    main()
  File "train.py", line 391, in main
    eval_metrics = validate(model, loader_eval, validate_loss_fn, args)
  File "train.py", line 535, in validate
    output = model(input)
  File "/home/mklachko/miniconda2/envs/pt/lib/python3.7/site-packages/torch/nn/modules/module.py", line 541, in __call__
    result = self.forward(*input, **kwargs)
  File "/home/mklachko/pytorch-image-models/timm/models/efficientnet.py", line 276, in forward
    x = self.forward_features(x)
  File "/home/mklachko/pytorch-image-models/timm/models/efficientnet.py", line 269, in forward_features
    x = self.blocks(x)
  File "/home/mklachko/miniconda2/envs/pt/lib/python3.7/site-packages/torch/nn/modules/module.py", line 541, in __call__
    result = self.forward(*input, **kwargs)
  File "/home/mklachko/miniconda2/envs/pt/lib/python3.7/site-packages/torch/nn/modules/container.py", line 92, in forward
    input = module(input)
  File "/home/mklachko/miniconda2/envs/pt/lib/python3.7/site-packages/torch/nn/modules/module.py", line 541, in __call__
    result = self.forward(*input, **kwargs)
  File "/home/mklachko/miniconda2/envs/pt/lib/python3.7/site-packages/torch/nn/modules/container.py", line 92, in forward
    input = module(input)
  File "/home/mklachko/miniconda2/envs/pt/lib/python3.7/site-packages/torch/nn/modules/module.py", line 541, in __call__
    result = self.forward(*input, **kwargs)
  File "/home/mklachko/pytorch-image-models/timm/models/efficientnet_blocks.py", line 262, in forward
    x = self.conv_pw(x)
  File "/home/mklachko/miniconda2/envs/pt/lib/python3.7/site-packages/torch/nn/modules/module.py", line 541, in __call__
    result = self.forward(*input, **kwargs)
  File "/home/mklachko/miniconda2/envs/pt/lib/python3.7/site-packages/torch/nn/modules/conv.py", line 345, in forward
    return self.conv2d_forward(input, self.weight)
  File "/home/mklachko/miniconda2/envs/pt/lib/python3.7/site-packages/torch/nn/modules/conv.py", line 342, in conv2d_forward
    self.padding, self.dilation, self.groups)
  File "/home/mklachko/miniconda2/envs/pt/lib/python3.7/site-packages/apex/amp/wrap.py", line 28, in wrapper
    return orig_fn(*new_args, **kwargs)
RuntimeError: cuDNN error: CUDNN_STATUS_NOT_SUPPORTED. This error may appear if you passed in a non-contiguous input.
Traceback (most recent call last):
  File "/home/mklachko/miniconda2/envs/pt/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/mklachko/miniconda2/envs/pt/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/mklachko/miniconda2/envs/pt/lib/python3.7/site-packages/torch/distributed/launch.py", line 253, in <module>
    main()
  File "/home/mklachko/miniconda2/envs/pt/lib/python3.7/site-packages/torch/distributed/launch.py", line 249, in main
    cmd=cmd)
subprocess.CalledProcessError: Command '['/home/mklachko/miniconda2/envs/pt/bin/python', '-u', 'train.py', '--local_rank=0', '/scratch/mklachko/data/imagenet/', '--model', 'efficientnet_b0', '-b', '512', '--sched', 'step', '--epochs', '450', '--decay-epochs', '2.4', '--decay-rate', '.97', '--opt', 'rmsproptf', '--opt-eps', '.001', '-j', '8', '--warmup-lr', '1e-6', '--weight-decay', '1e-5', '--drop', '0.2', '--drop-connect', '0.2', '--model-ema', '--model-ema-decay', '0.9999', '--aa', 'rand-m9-mstd0.5', '--remode', 'pixel', '--reprob', '0.2', '--amp', '--lr', '.032', '--output', 'output_B0', '--log-interval', '1000', '-j', '8']' returned non-zero exit status 1.
```

Same code when used on 4 Titan Xp cards using batch size 128 runs fine. 

Pytorch 1.3.1
NVIDIA Driver Version: 418.67       
CUDA Version: 10.1
torch.backends.cudnn.version(): 7603

**Comment by ngimel** [comment]
I'm closing this issue, dumping all CUDNN_STATUS_NOT_SUPPORTED errors here it not helpful. When encountering this error, please open new issue (unless you are sure that your case is related to the existing), and also try to make your bug report more actionable by isolating the call that causes an error. Try running with CUDA_LAUNCH_BLOCKING=1 and enable cudnn logginghttp://docs.nvidia.com/deeplearning/sdk/cudnn-developer-guide/index.html#api-logging

**Comment by michaelklachko** [comment]
Has this issue been fixed?  I'm talking about CUDNN error that happens for convolution with large matrix sizes - as is clearly shown in my error message. If so, what is the build I should test it with?

**Comment by ptrblck** [comment]
@michaelklachko Your issue might be related to the size limitations for cudnn convs as tracked in [this issue](https://github.com/pytorch/pytorch/issues/22496). @zasdfgbnm is working on a fix.

Let me know, if you think it's unrelated to your issue.

**Comment by michaelklachko** [comment]
Maybe: 3x224x224x64x512 > 2**32. It’s a pretty standard input size though, so I’m not sure why this isn’t a more widespread problem.

**Comment by ngimel** [comment]
That would fail in training though, not after the first epoch. Or do you use large batch size of eval only? It would be helpful if you isolated the failing call with the help of cudnn logging.

**Comment by michaelklachko** [comment]
Good point. It should be pretty easy for whoever is working on this issue to reproduce it by cloning the repo, and running the command on a V100 card. I'm not really familiar with cudnn debugging. Unless it's really simple, in that case please provide the detailed steps of what I need to do.

**Comment by richard-vogl** [comment]
> I'm closing this issue, dumping all CUDNN_STATUS_NOT_SUPPORTED errors here it not helpful. When encountering this error, please open new issue (unless you are sure that your case is related to the existing), and also try to make your bug report more actionable by isolating the call that causes an error. Try running with CUDA_LAUNCH_BLOCKING=1 and enable cudnn logginghttp://docs.nvidia.com/deeplearning/sdk/cudnn-developer-guide/index.html#api-logging

OK so should I open a new issue? I provided a isolated snipped in a comment above:

`import torch
x = torch.rand(1, 16, 70000, 100).to('cuda:0')
conv = torch.nn.Conv2d(in_channels=16, out_channels=32, kernel_size=(3,3), bias=False).to('cuda:0')
out = conv(x)`

i can even reproduce the issue using a google colab notebook:

https://colab.research.google.com/drive/1szd95jnY0rkcblIq8n9Zo0BpsWdFhrf3

So it is hard to imagine that the issue cannot reproduced.
I also offered to create an account on a machine where the issue can be reproduced for debugging purposes...

**Comment by ngimel** [comment]
Please file a new issue, tag @ptrblck on it.

**Comment by mlizhardy** [comment]
I downgraded to torch version 1.1.0 and my error went away "pip install torch==1.1.0"

**Comment by sjtuytc** [comment]
I resolve it by adding contiguous() after permute(). This is NOT only a cuddn issue. See https://www.gitmemory.com/issue/pytorch/pytorch/32395/577134996.

**Comment by ptrblck** [comment]
@sjtuytc could you post a code snippet to reproduce this issue?
The linked code snippet runs fine on `1.8.0a0+17f8c32` and doesn't produce the error.
Also, all other provided code snippets work in this version.
https://github.com/pytorch/pytorch/issues/4107#issuecomment-350906473
https://github.com/pytorch/pytorch/issues/4107#issuecomment-352937808
https://github.com/pytorch/pytorch/issues/4107#issuecomment-380200677
https://github.com/pytorch/pytorch/issues/4107#issuecomment-544739942
https://github.com/pytorch/pytorch/issues/4107#issuecomment-558491584

**Comment by sjtuytc** [comment]
I'm using 1.3.0 because I only have cuda 10.0.

**Comment by ptrblck** [comment]
@sjtuytc The latest stable release binaries (`1.7`) ship with CUDA9.2, 10.1, 10.2, and 11.0.
You can find the install instructions [here](https://pytorch.org/get-started/locally/).

Also note, that you would only need to install the appropriate NVIDIA driver, if you want to run native PyTorch workloads.
A local CUDA installation is necessary, if you want to build from source or use custom CUDA extensions, which have to be compiled.

**Comment by ngimel** [comment]
Closing based on @ptrblck comment. Please don't reopen, if you have new repros on the latest builds, please open new issues.
